cache/demux.go

```diff
@@ -137,27 +134,24 @@ func (d *demux) resyncLaggingWatchers() {
 	defer d.mu.Unlock()
 
 	for w, nextRev := range d.laggingWatchers {
-		var oldestRev int64
-		if oldestEvent := d.history.PeekOldest(); oldestEvent != nil {
-			oldestRev = oldestEvent.Kv.ModRevision
-		}
+		oldestRev := d.history.PeekOldest()
 
 		if oldestRev != 0 && nextRev < oldestRev {
 			w.Stop()
 			delete(d.laggingWatchers, w)
 			continue
 		}
 		// TODO: re-enable key‐predicate in Filter when non‐zero startRev or performance tuning is needed
-		missedEvents := d.history.Filter(AfterRev(nextRev))
+		missed := d.history.Filter(AfterRev(nextRev))
 
-		for _, event := range missedEvents {
-			if !w.enqueueEvent(event) { // buffer overflow: watcher still lagging
+		for _, eventBatch := range missed {
```

Comment:
Cannot make a suggestion because github sucks.
```
		enqueueFailed := false
		for _, eventBatch := range missed {
			if !w.enqueueEvent(eventBatch) { // buffer overflow: watcher still lagging
				enqueueFailed = true
				break
			}
			nextRev = eventBatch[0].Kv.ModRevision + 1
		}

		if !enqueueFailed {
```